Documentation/CraftingNuGetPackages.md

```diff
@@ -0,0 +1,69 @@
+
+## Quickstart Summary
+
+Through the use of a [Build Targets or Props](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#include-msbuild-props-and-targets-in-a-package) file inside a NuGet package, we can leverage properties provided by MSBuildForUnity to appropriately select the contents to bring into the Unity project. The following snippet is taken from [Microsoft.Windows.MixedReality.DotNetWinRT](https://www.nuget.org/packages/Microsoft.Windows.MixedReality.DotNetWinRT/) NuGet package. It brings in the `Unity` folder packaged in the root of the NuGet package to the output folder inside Unity.
+
+```xml
+<Project>
+  <PropertyGroup>
+    <!--All extra resources for this package will be placed under this folder in the output directory.-->
+    <PackageDestinationFolder>$(MSBuildThisFileName)</PackageDestinationFolder>
+  </PropertyGroup>
+  
+  <ItemGroup Condition="'$(MSBuildForUnityVersion)' != ''">
+    <Content Include="$(MSBuildThisFileDirectory)..\Unity\**">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <!-- Don't show .meta files in Solution Explorer - it's not useful. -->
+      <Visible Condition="'%(Extension)' == '.meta'">false</Visible>
+      <Link>$(PackageDestinationFolder)\%(RecursiveDir)%(Filename)%(Extension)</Link>
+    </Content>
+  </ItemGroup>
+</Project>
+```
+
+## Understanding MSBuildForUnity Behavior
+
+MSBuildForUnity is based on MSBuild and thereby leverages the build rules including package and project reference resoluton of MSBuild. These steps are invoked at Editor Compilation stage with a set of MSBuildForUnity provided variables. At this stage, we have the following information available:
+
+- Unity Version (ie Major.Minor)
+- Platform being targeted (UWP, Android, Standalone, etc).
+- .NET Target Framework (4.6, .NET Standard 2.0)
+- Compilation define symbols specified
+- For UWP Platform we know Min and Target UWP SDK
+
+Because this stage is not aware of the target device, and happens before we build the player output (to Android Studio, XCode or Visual Studio) we don't have the following information:
+
+- The target platform of the device (ARM, x86, x64)
+
+This means that we don't know the target platform of the device and thereby can't bring in a specific native DLL (ARM, x86, x64). Thus for NuGet packages with Native code, we must add additional logic to bring in every DLL as well as Unity specific .meta files.
+
+### Meta Files
+
+MSBuildForUnity performs meta file generation for .NET libraries, and there are plans to support processing Native DLLs as well, however this is not possible yet. For this reasons, packages with Native DLLs that must be brought into Unity must be accompanied with .meta files configured for those packages.
```

Comment:
" support processing Native DLLs as well, however this is not possible yet" - it possible to link to an issue with more details on this?

We have a number of cases in the MRTK where docs got stale because we said "we didnt' support this yet" but then added support later, and didn't update the docs. There are two reasons why I mention this:

1) If we have an issue tracking this, we can add more context for what this is (i.e. actually expand on what the feature is, and maybe something about how we might do it)
2) Even if we forget to update the docs later, someone could click on the issue number and see that it's actually resolved.